src/commands/partial/new.ts

```diff
@@ -0,0 +1,260 @@
+import * as path from "node:path";
+
+import { Flags } from "@oclif/core";
+import { prompt } from "enquirer";
+
+import BaseCommand from "@/lib/base-command";
+import { KnockEnv } from "@/lib/helpers/const";
+import { resolveResourceDir } from "@/lib/helpers/project-config";
+import { slugify } from "@/lib/helpers/string";
+import { promptToConfirm, spinner } from "@/lib/helpers/ux";
+import * as Partial from "@/lib/marshal/partial";
+import { PartialType } from "@/lib/marshal/partial";
+import {
+  ensureResourceDirForTarget,
+  PartialDirContext,
+  ResourceTarget,
+} from "@/lib/run-context";
+
+import PartialPush from "./push";
+
+const PARTIAL_TYPE_CHOICES = [
+  { name: PartialType.Html, message: "HTML" },
+  { name: PartialType.Json, message: "JSON" },
+  { name: PartialType.Markdown, message: "Markdown" },
+  { name: PartialType.Text, message: "Text" },
+] as const;
+
+export default class PartialNew extends BaseCommand<typeof PartialNew> {
+  static summary = "Create a new partial with a minimal configuration.";
+
+  static flags = {
+    name: Flags.string({
+      summary: "The name of the partial",
+      char: "n",
+    }),
+    key: Flags.string({
+      summary: "The key of the partial",
+      char: "k",
+    }),
+    type: Flags.string({
+      summary:
+        "The type of the partial (html, json, markdown, text). You cannot use this flag with --template.",
+      char: "t",
+      options: [
+        PartialType.Html,
+        PartialType.Json,
+        PartialType.Markdown,
+        PartialType.Text,
+      ],
+    }),
+    environment: Flags.string({
+      summary:
+        "The environment to create the partial in. Defaults to development.",
+      default: KnockEnv.Development,
+    }),
```

Comment:
This should also support the `branch` flag, correct?


```suggestion
    environment: Flags.string({
      summary:
        "The environment to create the partial in. Defaults to development.",
      default: KnockEnv.Development,
    }),
    branch: CustomFlags.branch,
```